Sources/ArrayExtensions.swift

```diff
@@ -201,6 +203,16 @@ extension Array where Element: Equatable {
     }
 }
 
+extension Array where Element: Hashable {
+
+    /// EZSE: Removes all occurrences of the given object(s)
+    public mutating func removeAll(_ elements: [Element]) {
+        let elementsSet = Set(elements)
```

Comment:
I might still be overlooking something, but note that the `Array` to `Set` transformation is never applied to the caller (i.e., `self`), it is only applied to the supplied `elements` array, as there is no value having duplicates in the array that describes which elements of `self` that are to be removed. For the caller (`self`), an `O(n)` filter is applied onto the `self` `Array`, and only the predicate in the `filter` closures makes use of `Hashable` logic for the supplied `elements`.

To show in practice, the following addition to `testRemoveObjectsByArray()` would cover your example above, and asserts to `true`:

```
var numberArrayWithDupes = [1, 1, 2, 2, 3]
let compareArrayForDupes = [1, 1, 2, 2]
numberArrayWithDupes.removeAll([3]) // use Hashable's removeAll(_:[Elements])
XCTAssertEqual(numberArrayWithDupes, compareArrayForDupes)
```

Finally, note that calling `removeAll(_:)` by variadic arguments (e.g. `arr.removeAll(3)` as in your comment above) will always call the single `Equatable` variadic overload. As I write in the description above, a call with variadic arguments will always be limited in size, in which case making using of `O(1)` lookup for `Hashable` elements, specifically, have limited value.